component/messageHandler.go

```diff
@@ -0,0 +1,96 @@
+package component
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	p_buff "th2-grpc/th2_grpc_common"
+
+	"github.com/rs/zerolog/log"
+	rabbitmq "github.com/th2-net/th2-common-go/schema/modules/mqModule"
+	"github.com/th2-net/th2-common-go/schema/queue/MQcommon"
+)
+
+type MessageTypeListener struct {
+	MessageType    string
+	Function       func(args ...interface{})
+	RootEventID    *p_buff.EventID
+	Module         *rabbitmq.RabbitMQModule
+	AmountReceived *int
+	NBatches       int
+	Stats          map[string]int
+}
+
+func NewListener(RootEventID *p_buff.EventID, module *rabbitmq.RabbitMQModule, BoxConf *BoxConfiguration, Function func(args ...interface{})) *MessageTypeListener {
+	amountReceived := 0
+	return &MessageTypeListener{
+		MessageType:    BoxConf.MessageType,
+		Function:       Function,
+		RootEventID:    RootEventID,
+		Module:         module,
+		AmountReceived: &amountReceived,
+		NBatches:       4,
+		Stats:          make(map[string]int),
+	}
+}
+
+func (listener MessageTypeListener) Handle(delivery *MQcommon.Delivery, batch *p_buff.MessageGroupBatch,
+	confirm *MQcommon.Confirmation) error {
+
+	defer func() {
+		if r := recover(); r != nil {
+			log.Err(fmt.Errorf("%v", r)).Msg("Error occurred while processing the received message.")
+		}
```

Comment:
Could we move this handler into main method and close common factory when panic happens?

Reply:
NOTE: common factory should close all subscription gracefully